modules/features/settings/src/main/java/au/com/shiftyjelly/pocketcasts/settings/privacy/UserAnalyticsSettings.kt

```diff
@@ -13,25 +13,24 @@ class UserAnalyticsSettings @Inject constructor(
     fun updateAnalyticsSetting(enabled: Boolean) {
         if (enabled) {
             settings.collectAnalytics.set(true, updateModifiedAt = true)
-            settings.collectAnalyticsThirdParty.set(true, updateModifiedAt = true)
             analyticsTracker.track(AnalyticsEvent.ANALYTICS_OPT_IN)
         } else {
             analyticsTracker.track(AnalyticsEvent.ANALYTICS_OPT_OUT)
             settings.collectAnalytics.set(false, updateModifiedAt = false)
-            settings.collectAnalyticsThirdParty.set(false, updateModifiedAt = false)
             analyticsTracker.clearAllData()
         }
     }
 
     fun updateAnalyticsThirdPartySetting(enabled: Boolean) {
         if (enabled) {
-            settings.collectAnalyticsThirdParty.set(true, updateModifiedAt = true)
+            settings.collectAnalyticsThirdParty.set(value = true, updateModifiedAt = true)
             analyticsTracker.track(AnalyticsEvent.ANALYTICS_THIRD_PARTY_OPT_IN)
         } else {
             analyticsTracker.track(AnalyticsEvent.ANALYTICS_THIRD_PARTY_OPT_OUT)
-            settings.collectAnalyticsThirdParty.set(false, updateModifiedAt = false)
+            settings.collectAnalyticsThirdParty.set(value = false, updateModifiedAt = true)
```

Comment:
I'm not 100% sure about this, but I believe we should set the `updateModifiedAt` in case we ever sync this setting with the server. If we update the modified at, we will know the user's last choice by the date set.